src/core/server/plugins/plugins_system.ts

```diff
@@ -167,54 +168,18 @@ export class PluginsSystem {
     return contracts;
   }
 
-  private async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
+  public async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
     // make _cat/plugins?format=json call to the OpenSearch instance
-    const opensearchInstalledPlugins = await this.getOpenSearchPlugins(deps);
+    const opensearchInstalledPlugins = await this.crossCompatibilityService.getOpenSearchPlugins(
+      deps.opensearch
+    );
     for (const pluginName of this.satupPlugins) {
       this.log.debug(`For plugin "${pluginName}"...`);
       const plugin = this.plugins.get(pluginName)!;
-      const pluginOpenSearchDeps = Object.entries(plugin.requiredEnginePlugins);
-      for (const [enginePluginName, versionRange] of pluginOpenSearchDeps) {
-        // add check to see if the installing Dashboards plugin version is compatible with installed OpenSearch plugin
-        if (
-          !this.isVersionCompatibleOSPluginInstalled(
-            opensearchInstalledPlugins,
-            enginePluginName,
-            versionRange
-          )
-        ) {
-          this.log.warn(
-            `OpenSearch plugin "${enginePluginName}" is not installed on the engine for the OpenSearch Dashboards plugin to function as expected.`
-          );
-        }
-      }
-    }
-  }
-
-  private isVersionCompatibleOSPluginInstalled(
-    opensearchInstalledPlugins: CatPluginsResponse,
-    depPlugin: string,
-    versionRange: string
-  ) {
-    return opensearchInstalledPlugins.find(
-      (obj) =>
-        obj.component === depPlugin &&
-        semver.satisfies(semver.coerce(obj.version)!.version, versionRange)
-    );
-  }
-
-  private async getOpenSearchPlugins(deps: PluginsServiceStartDeps) {
-    // Makes cat.plugin api call to fetch list of OpenSearch plugins installed on the cluster
-    try {
-      const { body } = await deps.opensearch.client.asInternalUser.cat.plugins<any[]>({
-        format: 'JSON',
-      });
-      return body;
-    } catch (error) {
-      this.log.warn(
-        `Cat API call to OpenSearch to get list of plugins installed on the cluster has failed: ${error}`
+      await this.crossCompatibilityService.checkPluginVersionCompatibility(
```

Comment:
So the health check is not doing anything here other than recording the compatibility for itself. What are the plans to notify the OSD plugins if one of their deps becomes incompatible?

Reply:
Yes that's correct. To notify the OSD plugins if one of their deps becomes incompatible during runtime needs a hook to be identified to embed this new service when engine state changes. For now, plugins can call the cross compatibility service API on their own during runtime to get to know the compatibility state.

Reply:
Let's create an issue for an enhancement which will have plugins possibly expose a function that we call when compatibility changes.

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5279